server/authorizer/authorizer_test.go

```diff
@@ -226,3 +227,42 @@ func TestBasicEscapeError(t *testing.T) {
 		t.Fatalf("Expected invalid parameter and URL parse error but got: %v", recorder)
 	}
 }
+
+func TestMakeInput(t *testing.T) {
+	path := "/data/some/specific/document?pretty=true&explain=\"full\""
+	req, err := http.NewRequest(http.MethodGet, "http://localhost:8181"+path, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	query := req.URL.Query()
+
+	// set query parameters
+	query.Set("pretty", "true")
+	query.Set("explain", "full")
+	req.URL.RawQuery = query.Encode()
+
+	identity := identifier.Identity(req)
+
+	parsedPath, err := parsePath(req.URL.Path)
+	if err != nil {
+		panic(err)
+	}
+
+	result, err := makeInput(req)
+	if err != nil {
+		panic(err)
+	}
+
+	expectedResult := map[string]interface{}{
+		"path":     parsedPath,
+		"method":   strings.ToUpper(req.Method),
+		"identity": identity,
+		"params":   query,
+	}
+
+	if !reflect.DeepEqual(result, expectedResult) {
```

Comment:
I think it would be better to compare the input against the AST or JSON representation of the input. As is, it's not easy to tell that the query params are being serialized correctly.

For example:

```golang
expected := util.MustUnmarshalJSON([]byte(`
{
  "path": ["foo","bar"],
  "method": "GET",
  "identity": "bob",
  "params": {"pretty": ["true"], "explain": ["full"]}
}
`)

if !reflect.DeepEqual(result, expected) { // error }
```